caplena/endpoints/projects_endpoint.py

```diff
@@ -251,7 +286,110 @@ def update_row(
 # --- Resources & Objects--- #
 
 
-class ProjectDetail(BaseResource[ProjectsController]):
+class OperationsProtocol(Protocol):
+    # controller: ProjectsController
+    @property
+    def controller(self) -> ProjectsController:
+        ...
+
+    @property
+    def id(self) -> str:
+        ...
+
+    def _refresh_from(self, *, attrs: Dict[str, Any]) -> None:
+        ...
+
+    def modified_dict(self) -> Any:
+        ...
+
+
+class RowOperationsMixin(OperationsProtocol, Protocol):
```

Comment:
Curiousity question:why having Protocol here too? Isn't OperationsProtocol already a Protocol?

Reply:
When you inherit from class which inherited from protocol It's not a protocol anymore as it's considered as implementation already, this is why It's again inheriting from protocol. :)

Reply:
TIL .... also found https://mypy.readthedocs.io/en/stable/protocols.html#defining-subprotocols-and-subclassing-protocols which explains it.